syft/grid/clients/data_centric_fl_client.py

```diff
@@ -284,3 +285,77 @@ def delete_model(self, model_id: str) -> bool:
 
     def __str__(self) -> str:
         return f"<Federated Worker id:{self.id}>"
+
+    @staticmethod
+    def simplify(worker: AbstractWorker, data_centric_fl_client: "DataCentricFLClient") -> tuple:
+
+        # Simplify the attributes for DataCentricFLClient
+        address = pickle.dumps(data_centric_fl_client.address)
+        id = pickle.dumps(data_centric_fl_client.id)
+        is_client_worker = pickle.dumps(data_centric_fl_client.is_client_worker)
+        log_msgs = pickle.dumps(data_centric_fl_client.log_msgs)
+        verbose = pickle.dumps(data_centric_fl_client.verbose)
+        encoding = pickle.dumps(data_centric_fl_client.encoding)
+        timeout = pickle.dumps(data_centric_fl_client.timeout)
+
+        return (address, id, is_client_worker, log_msgs, verbose, encoding, timeout)
+
+    @staticmethod
+    def detail(
+        worker: AbstractWorker, client_tuple: tuple
+    ) -> Union["DataCentricFLClient", int, str]:
+
+        address, id, is_client_worker, log_msgs, verbose, encoding, timeout = client_tuple
+
+        # detail client attributes
+        address = pickle.loads(address)
+        id = pickle.loads(id)
+        is_client_worker = pickle.loads(is_client_worker)
+        log_msgs = pickle.loads(log_msgs)
+        verbose = pickle.loads(verbose)
+        encoding = pickle.loads(encoding)
+        timeout = pickle.loads(timeout)
+
+        # create torch hook
```

Comment:
Q: if we reach here doesn't mean we already are hooked?

Reply:
yup we have created but I am not sure how to access that hook from here, also creating a new one should not raise any issue.

Reply:
I think you can access it from the worker.

Reply:
The problem is that `detail` is called by a gridnode on the simplified objects it receives over the network from client(so we don't have control from client side over the behaviour of the function.). 
and you cant pass the hook to it, (unless I serialise the hook instance in `simplify ` function which we will mean we are essentially sending hook over the network which is not required.

Reply:
and it's a class method which takes simplified input as arguments.

Reply:
It can't be access using ```syft.local_worker.hook```?

Reply:
@Nilanshrajput I think that @gmuraru is right. Please make this modification before merging.

Reply:
got it, updated